netket/nn/symmetric_linear.py

```diff
@@ -211,35 +572,213 @@ def full_bias(self, bias):
+    else:
+        raise ValueError(
+            f"Unknown mode={mode}. Valid modes are 'fft', 'matrix', or 'auto'."
+        )
+
+
+def DenseEquivariant(symmetries, mode="auto", shape=None, point_group=None, **kwargs):
+    r"""A group convolution operation that is equivariant over a symmetry group.
+
+    Acts on a feature map of symmetry poses of shape [batch_size,n_symm*in_features]
+    and returns a feature  map of poses of shape [batch_size,n_symm*out_features]
+
+    G-convolutions are described in ` Cohen et. {\it al} <http://proceedings.mlr.press/v48/cohenc16.pdf>`_
+    and applied to quantum many-body problems in ` Roth et. {\it al} <https://arxiv.org/pdf/2104.05085.pdf>`_
+
+    The G-convolution generalizes the convolution to non-commuting groups:
+
+    .. math ::
+
+        C^i_g = \sum_h {\bf W}_{g^{-1} h} \cdot {\bf f}_h
+
+    Group elements that differ by the same symmetry operation (i.e. :math:`g = xh`
+    and :math:`g' = xh'`) are connected by the same filter.
+
+    Args:
+        symmetries: A specification of the symmetry group. Can be given by a
+            nk.graph.Graph, an nk.utils.PermuationGroup, a list of irreducible
+            representations or a product table
+        mode: string "fft, irreps, matrix, auto" specifying whether to use a fast
+            fourier transform over the translation group, a fourier transform using
+            the irreducible representations or by constructing the full kernel matrix.
+        features: The number of symmetry-reduced features. The full output size
+            is n_symm*features.
+        use_bias: A bool specifying whether to add a bias to the output (default: True).
+        mask: An optional array of shape [n_sites] consisting of ones and zeros
+            that can be used to give the kernel a particular shape
+        dtype: The datatype of the weights. Defaults to a 64bit float
+        precision: Optional argument specifying numerical precision of the computation
+            see `jax.lax.Precision`for details.
+        kernel_init: Optional kernel initialization function. Defaults to variance scaling
+        bias_init: Optional bias initialization function. Defaults to zero initialization
+    """
+
+    if isinstance(symmetries, Lattice) and (
+        not point_group is None or not symmetries._point_group is None
+    ):
+        shape = tuple(symmetries.extent)
+        # With graph try to find point group, otherwise default to automorphisms
+        sg = symmetries.space_group(point_group)
+        if mode == "auto":
+            mode = "fft"
+    elif isinstance(symmetries, Graph):
+        sg = symmetry_info.automorphisms()
+        if mode == "auto" or mode == "fft":
+            mode = "irreps"
+        if mode == "fft":
+            raise ValueError(
+                "When requesting 'mode=fft' a valid point group must be specified"
+                "in order to construct the space group"
+            )
```

Comment:
bug?

Reply:
```
        if mode == "auto" or mode == "fft":
            mode = "irreps"
```
you are changing a mode that was explicitly set.

Reply:
Yeah mode='fft' should be deleted from the conditional